src/main/java/org/grouplens/grapht/reflect/internal/ClassInstantiator.java

```diff
@@ -98,38 +106,51 @@ public Object instantiate() throws ConstructionException {
         } catch (IllegalAccessException e) {
             throw new ConstructionException(ctor, "Access violation on " + ctor, e);
         }
+        finally {
+            mdcContext.finish();
+        }
 
         // satisfy dependencies in the order of the list, which was
         // prepared to comply with JSR 330
         Map<Method, InjectionArgs> settersAndArguments = new HashMap<Method, InjectionArgs>();
         for (Desire d: desires) {
             if (d.getInjectionPoint() instanceof FieldInjectionPoint) {
+                LogContext mdcContextInjectionPointFieldValue = LogContext.create();
```

Comment:
Again, we need to start the `try` that closes this context right away. Also, we want to put the `put` statements right away, so that they are in effect when we instantiate the injection point.

The key point is that when we call `instantiate()` on the instantiator, down on line 120, the relevant value needs to be in the MDC.

I think that we can just use a single key, `org.grouplens.grapht.injectionPoint`, with the `toString()` of the injection point object, instead of separate keys. Injection points do a good job of describing themselves in `toString()`.